dan_layer/engine/tests/templates/nft/tickets/src/lib.rs

```diff
@@ -41,77 +41,63 @@ mod tickets {
     }
 
     impl TicketSeller {
-        pub fn new(initial_supply: u64, price: Amount, event_metadate: Metadata) -> Self {
+        // TODO: in this example we need to specify the payment resource, but there should be native support for Thaums
+        pub fn new(payment_resource_address: ResourceAddress, initial_supply: usize, price: Amount, event_description: String) -> Self {
             // Create the non-fungible resource
             // TODO: restrict minting to only the owner
             let resource_address = ResourceBuilder::non_fungible()
-                // The event metadata is common for all tickets, so it's stored only once in the resource, not in every ticket individually
-                .with_metadata(event_metadata)
+                // The event description is common for all tickets
+                .add_metadata("event", event_description)
                 .build();
 
             // Mint the initial tickets
-            let sample_ticket = Ticket::new();
-            let ticket_bucket = ResourceBuilder::get(resource_address)
+            let sample_ticket = NonFungible::new(Metadata::new(), &Ticket::default());
+            let ticket_bucket = ResourceManager::get(resource_address)
                 .mint_many_non_fungible(sample_ticket, initial_supply);
```

Comment:
Apologies, I have changed the nft interface slightly since you started this PR to avoid having a backend substate type (NonFungible) that contains the NFT data in WASM memory in the template_lib. `mint_non_fungible` now takes `nft_id, immutable_data: &impl Decode, mutable_data: &impl Decode` and there is an engine call to fetch the nft data as needed.

```suggestion
            let ticket_bucket = ResourceManager::get(resource_address)
                .mint_many_non_fungible(&Metadata::new(), &Ticket::default(), initial_supply);
```

Recommend merging development into this PR and fixing what breaks.